compiler/resolution/callDestructors.cpp

```diff
@@ -1027,6 +1027,13 @@ static void collectGlobals(ModuleSymbol* mod,
           if (VarSymbol* v = initsVariableOut(formal, actual))
             noteGlobalInitialization(mod, v, inited, initedSet);
         }
+
+        // Recurse into calls to compiler generated functions that
+        // return array types (forall expression functions).
+        // (Seach for globalTemps in normalize.cpp to see related code).
+        if (FnSymbol* calledFn = fCall->resolvedFunction())
+          if (calledFn->hasFlag(FLAG_MAYBE_ARRAY_TYPE))
```

Comment:
Does it make sense to make this more focused? Like check for `FLAG_COMPILER_GENERATED` at the very least?

Reply:
I don't think it's a big deal either way because the traversal will only look for initialization of globals. But I can add that check.

Reply:
Thanks for adding it. I am always a little worried with bandaids and their unforeseen consequences down the road, so I try to make them as specific as possible out of abundance of caution. (Maybe to the extend that they become ugly implementations  :) )